src/app/components/match/match.component.ts

```diff
@@ -19,13 +21,14 @@ export class MatchComponent implements OnInit {
   matchForm: FormGroup;
   date: NgbDateStruct;
 
-  time = { hour: 0, minute: 0 };
-  eventDate: Date = new Date();
+  event_Date: Date = new Date();
 
+  volunteeringEvent: VolunteeringEventModel;
   volunteeringRequests: VolunteeringRequestModel[] = [];
   volunteeringOffers: VolunteeringOfferModel[] = [];
-  selectedRequest: VolunteeringRequestModel = { _id: null } as VolunteeringRequestModel;
-  selectedOffer: VolunteeringOfferModel = { _id: null } as VolunteeringOfferModel;
+  plannedEvents: VolunteeringEventModel[];
+  selectedOffer: VolunteeringOfferModel = {} as any;
```

Comment:
Shouldn't the initial value here be `undefined`? (It's given if you don't assign)